shell/platform/embedder/embedder.h

```diff
@@ -989,6 +1056,25 @@ typedef struct {
   };
 } FlutterMetalBackingStore;
 
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterVulkanBackingStore).
+  size_t struct_size;
+  /// The Vulkan image that the layer will be rendered to. This image must
+  /// already be available for the engine to bind for writing when it's given to
+  /// the engine via the backing store creation callback. The engine will
+  /// perform a host sync for all layers prior to calling the compositor present
+  /// callback, and so the written layer images can be freely bound by the
+  /// embedder without any additional synchronization.
+  FlutterVulkanImage image;
```

Comment:
Embedding this struct directly seems very dangerous to me.  By doing this *any* changes to FlutterVulkanImage will break ABI -- that structure becomes impossible to change.

I think you could move the FlutterVulkanImage to the end of FlutterVulkanBackingStore, but as soon as someone adds a new field to FlutterVulkanBackingStore you'll end up back at square one re: ABI breaks.

@chinmaygarde

Reply:
Hmm, good catch. The only way I can think of to get around this completely is to just inline the fields of `FlutterVulkanImage` in `FlutterVulkanBackingStore`. Since it's so simple and shouldn't get much more complicated in the future, that seems like it'd be a fine solution. WDYT?

Reply:
I ended up switching this to a pointer to const FlutterVulkanImage [here](https://github.com/flutter/engine/pull/29391#discussion_r789035359).